core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -901,6 +916,282 @@ public TableToken lockTableName(CharSequence tableName, String dirName, int tabl
         return tableNameRegistry.lockTableName(tableNameStr, dirName, tableId, isWal);
     }
 
+    public void metadataCacheClear() {
+        cairoTables.clear();
+    }
+
+    /**
+     * Gets the current table metadata object from the list.
+     * <p>
+     * Returns a table if present.
+     * Returns null if table not present or could not be re-hydrated after a metadata change.
+     *
+     * @param tableToken A current table token for the table.
+     * @return The cached table metadata.
+     */
+    public CairoTable metadataCacheGetTable(@NotNull TableToken tableToken) {
+        final CairoTable table = cairoTables.get(tableToken.getDirName());
+        if (table == null) {
+            metadataCacheHydrateTable(tableToken, false, true);
+        }
+        return cairoTables.get(tableToken.getDirName());
+    }
+
+    /**
+     * This is dangerous and may clobber any concurrent metadata changes. This should only be used in last-resort cases,
+     * for example, for testing purposes.
+     */
+    public void metadataCacheHydrateAllTables() {
+        ObjHashSet<TableToken> tableTokensSet = tlTokens.get();
+        getTableTokens(tableTokensSet, false);
+        ObjList<TableToken> tableTokens = tableTokensSet.getList();
+        TableToken tableToken = tableTokens.getQuick(0);
+
+        try {
+            for (int i = 0, n = tableTokens.size(); i < n; i++) {
+                tableToken = tableTokens.getQuick(i);
+                if (!tableToken.isSystem()) {
+                    metadataCacheHydrateTable(tableTokens.getQuick(i), true, true);
+                }
+            }
+        } catch (CairoException ex) {
+            LOG.error().$("could not hydrate metadata: [table=").$(tableToken).I$();
```

Comment:
cairo exception usually contains error message and more importantly OS errorno. We have to always log both.